src/Core/Resolvers/Factories/QueryEngineFactory.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Net;
+using Azure.DataApiBuilder.Auth;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Core.Configurations;
+using Azure.DataApiBuilder.Core.Models;
+using Azure.DataApiBuilder.Core.Services.MetadataProviders;
+using Azure.DataApiBuilder.Service.Exceptions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.DataApiBuilder.Core.Resolvers.Factories
+{
+    /// <summary>
+    /// QueryEngineFactory class.
+    /// Used to get the appropriate queryEngine based on database type.
+    /// </summary>
+    public class QueryEngineFactory : IQueryEngineFactory
+    {
+        private readonly IEnumerable<IQueryEngine> _queryEngines;
```

Comment:
same comment as mutationEnginefactory -> could be dictionary? this is also private -> an implementation detail, so i don't see benefit of Ienumerable abstraction.

Reply:
fixed.